sklearn/neighbors/_lof.py

```diff
@@ -342,16 +354,7 @@ def predict(self):
         is_inlier : ndarray of shape (n_samples,)
             Returns -1 for anomalies/outliers and +1 for inliers.
         """
-        if not self.novelty:
-            msg = (
-                "predict is not available when novelty=False, use "
-                "fit_predict if you want to predict on training data. Use "
-                "novelty=True if you want to use LOF for novelty detection "
-                "and predict on new unseen data."
-            )
-            raise AttributeError(msg)
-
-        return self._predict
+        return self._predict(X)
 
     def _predict(self, X=None):
```

Comment:
Is there a reason for not merging `_predict` in `predict`

Reply:
I think we need to keep `_predict` for `fit_predict`:

https://github.com/scikit-learn/scikit-learn/blob/81165cabad383db2ff7fd856e467041eea9b55dc/sklearn/neighbors/_lof.py#L208-L264

Reply:
Oh I see. However, `fit_predict` should not be a `property` by the way :)

Reply:
I was planning to fix fit methods in a separate PR, but the number fit methods decorated by `property` is small so I'll fix them in this PR.